src/rgw/rgw_coroutine.h

```diff
@@ -542,7 +544,7 @@ class RGWCoroutinesManager {
   int run(list<RGWCoroutinesStack *>& ops);
   int run(RGWCoroutine *op);
   void stop() {
-    if (going_down.inc() == 1) {
+    if (!going_down) {
```

Comment:
the `inc()` here was important, allowing `stop()` to be called multiple times, but only calling `completion_mgr->go_down()` the first time

you -could- turn it back into an `atomic<unsigned>` and use `if (++going_down == 1)`, but i think [compare_exchange](http://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange) is a better fit for this:
```c++
bool expected = false;
if (going_down.compare_exchange_strong(expected, true)) {
  completion_mgr->go_down();
}
```